### PR TITLE
Clean out legacy MSSQL db_handler code

### PR DESCRIPTION
**Pull request by yocheah**
Please include a summary of the change and which issue is addressed.

Fixes / Closes / Implements / Issue #(issue) 

## PR Self Evaluation

- [x] I have reviewed my code to check that it contains no sensitive information.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests or modified existing tests that prove my fix is effective or that my feature works
- [x] Existing unit tests pass locally with my changes

### CODE REVIEW

File: processing/base_creator.py [hunk 1]
```diff
@@ -314,12 +308,12 @@ def driver(self):
...
         base_attrs = {}
         psql_conn = self.db_conn_pool.get('psql_conn')
-        self.site_list = self.flux_db_handler.get_sites_with_updates()
+        self.site_list = self.new_db_handler.get_sites_with_updates()
         self.embargoed_site_list = self.new_db_handler.get_sites_with_embargo(
             psql_conn, self.embargo_years)
         self.historic_site_list = self.new_db_handler.get_sites_with_updates(
             psql_conn, is_historic=True)
-        self.preBASE_files = self.flux_db_handler.get_BASE_candidates()
+        self.preBASE_files = self.new_db_handler.get_base_candidates()
```
Comment (dschristianson) [inline processing/base_creator.py:316]: method has parameter "state_ids".
Old code does not pass anything thru but I'm not see how that worked. There is no default value for state_ids in the method.

**Review by dschristianson — COMMENTED** [review]
question.

File: processing/base_creator.py (hunk 1 above)
Comment (yocheah) [reply processing/base_creator.py:316]: Thanks. Looks like you updated the original definition on March 18th, but the base_creator code was not updated somehow.

Comment (yocheah) [reply processing/base_creator.py]: I just pushed a fix, should make use of the newer ProcessStatesHandler class

File: processing/db_handler.py [hunk 2]
```diff
@@ -134,6 +134,113 @@ def get_base_badm_update_zip_count(self, conn):
...
+    #                 row.get('base_version'), row.get('process_id'))
+    #     return site_ids
+
+    def define_base_candidates_query(
+            self, pre_query, post_query, state_ids):
+        full_query_components = [pre_query]
+        query_criteria = SQL('s.state_log = %s ')
+        for idx, state_id in enumerate(state_ids):
+            if idx > 0:
+                full_query_components.append(SQL( 'OR '))
```
Comment (dschristianson) [inline processing/db_handler.py:182]: For state_id elements after the first one, looks like only 'OR' is being added. The query criteria also needs to be added.

File: processing/db_handler.py [hunk 3]
```diff
@@ -134,6 +134,113 @@ def get_base_badm_update_zip_count(self, conn):
...
+                'GROUP BY process_id) latest_state '
+                'ON latest_state.latest_state_timestamp = s.log_timestamp '
+                'AND s.process_id = latest_state.process_id) s '
+                'ON s.process_id = l.log_id '
+                'INNER JOIN qaqc.process_summarized_output o '
+                'ON o.process_id = l.log_id '
+                'LEFT JOIN qaqc.publishing_log p '
+                'ON p.process_id = l.log_id '
+                'WHERE ')
+            post_query = SQL(') s ON s.process_id = l.log_id')
```
Comment (dschristianson) [inline processing/db_handler.py:212]: Is this post query already included in the pre_query -- line 206?

**Review by dschristianson — COMMENTED** [review]
I can't get the get_base_candidate query to work. See notes inline.

File: processing/update_base_badm.py [hunk 4]
```diff
@@ -237,7 +231,9 @@ def driver(self, base_attrs, post_base_only=None):
         badm_map = self.new_db_handler.get_badm_map(psql_conn)
         sites_needing_updates = self.new_db_handler.get_sites_with_updates(
             psql_conn)
-        base_candidate_map = self.flux_db_handler.get_BASE_candidates(
+
+        # Need to revamp
```
Comment (dschristianson) [inline processing/update_base_badm.py:235]: Consider removing comment unless there is still revamping to be completed?

**Review by dschristianson — APPROVED** [review]
Minor comment -- approving as is.

File: processing/update_base_badm.py (hunk 4 above)
Comment (yocheah) [reply processing/update_base_badm.py]: Will do

**Pull request merged**